flytekit/core/type_engine.py

```diff
@@ -437,7 +447,29 @@ def to_literal_type(cls, python_type: Type) -> LiteralType:
         Converts a python type into a flyte specific ``LiteralType``
         """
         transformer = cls.get_transformer(python_type)
-        return transformer.get_literal_type(python_type)
+        res = transformer.get_literal_type(python_type)
+        data = None
+        if typing.get_origin(python_type) is typing.Annotated:
+            for x in typing.get_args(python_type)[1:]:
+                if not isinstance(x, FlyteAnnotation):
+                    continue
+                if x.data.get("__consumed", False):
+                    continue
```

Comment:
can we add a comment as to why we need a check at all?  If we're just doing a straight loop, why would we ever be in danger of reading something more than once?